client/src/components/FileViewer/index.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import ReactMarkdown from 'react-markdown';
+
+class FileViewer extends React.Component {
+
+  constructor (props) {
+    super(props);
+    /*
+      Prevent memory leak by closing fetch before unmount
+     */
+    this.abortController = new AbortController();
+    this.abortSignal = this.abortController.signal;
+    this.state = {
+      fileLoaded: false,
+      fileText  : '',
+    };
+  }
+
+  componentDidMount () {
+    const {sourceUrl} = this.props;
+    const signal = this.abortSignal;
+    fetch(sourceUrl, { signal })
+      .then(response => response.text())
+      .then((text) => {
+        this.setState({fileText: text});
+        this.setState({fileLoaded: true});
+        return true;
+      })
+      .catch(e => { console.log('fetch aborted on unmount ', e) });
+  }
```

Comment:
The signal /abortController piece prevents memory leak from not resolving a fetch promise on unmount. Currently the way I catch the error throws a console log in the browser. Maybe there's a better way.